django/db/models/sql/query.py

```diff
@@ -1357,9 +1357,6 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
                     clause.add(lookup_class(value, False), AND)
         return clause, used_joins if not require_outer else ()
 
-    def add_filter(self, filter_clause):
-        self.add_q(Q(**{filter_clause[0]: filter_clause[1]}))
-
```

Comment:
Once upon a time, this was a big old important method, back in 9c52d56f6f8a9cdafb231adf9f4110473099c9b5. The last time it had _any_ complexity of note was ecaba3602837d1e02fe1e961f7d3bf9086453259 I think.

Reply:
Maybe we can keep it as a tiny wrapped to avoid `self.add_q(Q(...))` in so many cases, `self.add_filter(lhs, rhs)` is much more clearer, IMO:
```python
def add_filter(self, filter_lhs, filter_rhs):
    self.add_q(Q(filter_lhs, filter_rhs))
```
We could change `add_filter()`'s signature and replace all `add_q(Q(..))` in the first commit.

Reply:
Restored it as per the suggestion(ish, you were missing a pair of parens - which is a good enough argument for keeping it around and hiding that), and updated the `add_q` uses to call it instead. 

The only places `add_q` is now called directly are: `django.db.models.query.QuerySet._filter_or_exclude_inplace`and `django.db.models.query.QuerySet.complex_filter` ... both of which are _by intention_ pre-composing the Q object so _should_ not pass through the `add_filter` method.